src/services/AudiusBackend.js

```diff
@@ -1022,13 +1024,40 @@ class AudiusBackend {
     return null
   }
 
+  /**
+   * Retrieves the user's associated wallets from IPFS using the user's metadata CID and creator node endpoints
+   * @param {Object} user The user metadata which contains the CID for the metadata multihash
+   * @returns Object The associated wallets mapping of address to nested signature
+   */
+  static async fetchUserAssociatedSolWallets(user) {
+    const gateways = getCreatorNodeIPFSGateways(user.creator_node_endpoint)
+    const cid = user?.metadata_multihash ?? null
+    if (cid) {
+      const metadata = await fetchCID(
```

Comment:
shouldn't we only need to fetch this CID once? maybe we should follow the pattern on the profile page for collectibles and do this somewhere else:
https://github.com/AudiusProject/audius-client/blob/b8b852bd4a819694eb60ae27b82bad79bed4e79b/src/containers/profile-page/store/sagas.js#L56

Reply:
interesting, yeah it's odd bc fetching the metadata and storing it redux means that the format of the current user's metadata is different from all other users in the cache.
Specifically for the linked wallets - usually we only fetch just user's linked wallets from a discovery endpoint, bc we don't need the extra metadata for signature.
I personally like the idea of fetching before setting the new wallets which is where this is used vs on init fetching of user because it leads to less of a chance of messing up the user's metadata in between. but there're a couple places I could reduce the network calls for this. 
Thoughts?